sql/src/main/java/io/druid/sql/calcite/expression/builtin/CastOperatorConversion.java

```diff
@@ -140,12 +140,12 @@ private static DruidExpression castCharToDateTime(
       final SqlTypeName toType
   )
   {
-    // Cast strings to datetimes by parsin them from SQL format.
+    // Cast strings to datetimes by parsing them from SQL format.
     final DruidExpression timestampExpression = DruidExpression.fromFunctionCall(
         "timestamp_parse",
         ImmutableList.of(
             operand,
-            DruidExpression.fromExpression(DruidExpression.stringLiteral(dateTimeFormatString(toType))),
+            DruidExpression.fromExpression(DruidExpression.nullLiteral()),
```

Comment:
Does this mean the dateTimeFormat parameter is always null when casting? If so, the below timeZone parameter looks not necessary.

Reply:
It does, but the timezone is still useful, since null for the format really just means "use iso or sql format" and that may still need a time zone to interpret the value.

For example `timestamp_parse('2000-01-01 00:00:00', null, 'UTC')` and `timestamp_parse('2000-01-01 00:00:00', null, 'America/Los_Angeles')` return different values.

Reply:
I mean, timezone looks not used if formatString is null (https://github.com/druid-io/druid/blob/master/processing/src/main/java/io/druid/query/expression/TimestampParseExprMacro.java#L58-L61). Maybe we need to fix TimestampParseExprMacro to use timezone for null formatString as well?

Reply:
Oh, sorry, you're right. This is a bug in TimestampParseExprMacro. I pushed another commit fixing it to this PR.

Reply:
Thanks!